noir-projects/aztec-nr/aztec/src/context/public_context.nr

```diff
@@ -275,17 +275,43 @@ impl PublicContext {
         is_static_call: bool,
         is_delegate_call: bool
     ) -> [Field; RETURN_VALUES_LENGTH] {
-        let side_effect_counter = self.side_effect_counter;
-        // TODO get next value from output of `call_public_function_internal`
-        self.side_effect_counter += 1;
-
-        call_public_function_internal(
+        let item = call_public_function_internal(
             contract_address,
             function_selector,
             args_hash,
-            side_effect_counter,
+            self.side_effect_counter,
             is_static_call,
             is_delegate_call
-        )
+        );
+
+        assert_eq(item.public_inputs.start_side_effect_counter, self.side_effect_counter);
+        self.side_effect_counter = item.public_inputs.end_side_effect_counter + 1;
+
+        assert(contract_address.eq(item.contract_address));
+        assert(function_selector.eq(item.function_data.selector));
+
+        assert(args_hash == item.public_inputs.args_hash);
+
+        // Assert that the call context of the enqueued call generated by the oracle matches our request.
+        assert(item.public_inputs.call_context.is_delegate_call == is_delegate_call);
+        assert(item.public_inputs.call_context.is_static_call == is_static_call);
+
+        if (is_delegate_call) {
```

Comment:
The checks for the content of the callee's `call_context` with the caller are already in the kernel circuits. Can probably remove it here and in private context.